core/controller/src/main/scala/whisk/core/entitlement/Entitlement.scala

```diff
@@ -214,11 +206,37 @@ protected[core] abstract class EntitlementService(config: WhiskConfig)(
         grant
     }
 
+    // NOTE: explicit grants do not work with package bindings because the current model
+    // for authorization does not allow for a continuation to check that both the binding
+    // and the references package are both either implicitly or explicitly granted; this is
+    // accepted for the time being however because there exists no external mechanism to create
+    // explicit grants
+    protected def check(subject: Subject, right: Privilege, resource: Resource)(
```

Comment:
Got it, @rabbah want to rename this to `checkPrivilege` or thelike. Overloading `check` seems like both are equivalent in effects, while they aren't.